pkg/testframework/framework.go

```diff
@@ -19,6 +23,19 @@ var (
 	AsyncOperationTimeout = 5 * time.Minute
 )
 
+// Logger is an interface to report events from tests. It is implemented by
```

Comment:
I'm wondering if `framework.go` might grow into a kitchen sink of a file.  Maybe we can break this up now & encourage future growth into separate files to keep it tidy down the road.

Reply:
(I realize there isn't much here yet, so it could just be `/framework/logging.go` & `framework/resourceavailabilty.go` or something a bit like that.)

Reply:
I see you have logs file below already. Nice.  Was just thinking about `DumpObject`... hmm.

Reply:
Yep thats a good question. But I think it mainly depends on how beefy we want our e2e tests for the console operator. If it should just test each operator state(maybe some extra tests, scaling operator down to 0 and back to 1), I wouldn't worry about the scale of the file :)

Reply:
Probably related to the above comment, we have a set of other features to test, but uncertain how much it will grow over time (I dont expect tons).